File: ios/Classes/util/NativeGeofenceLogger.swift
import CoreLocation
import Foundation
import OSLog

/// Centralized logger for native iOS code
/// Logs to iOS unified logging system (OSLog) only
class NativeGeofenceLogger {
    private let logger: Logger
    private let category: String
    
    init(category: String) {
        self.category = category
        self.logger = Logger(subsystem: Constants.PACKAGE_NAME, category: category)
    }
    
    /// Log a debug message
    func debug(_ message: String) {
        logger.debug("\(message)")
    }
    
    /// Log an info message
    func info(_ message: String) {
        logger.info("\(message)")
    }
    
    /// Log a warning message
    func warning(_ message: String) {
        logger.warning("\(message)")
    }
    
    /// Log an error message
    func error(_ message: String) {
        logger.error("\(message)")
    }
}

File: android/src/main/kotlin/com/steffaanus/native_geofence/util/NativeGeofenceLogger.kt
package com.steffaanus.native_geofence.util

import android.util.Log

/**
 * Centralized logger for native Android code
 * Logs to Android Logcat only
 */
class NativeGeofenceLogger(private val tag: String) {
    
    /**
     * Log a debug message
     */
    fun d(message: String) {
        Log.d(tag, message)
    }
    
    /**
     * Log an info message
     */
    fun i(message: String) {
        Log.i(tag, message)
    }
    
    /**
     * Log a warning message
     */
    fun w(message: String) {
        Log.w(tag, message)
    }
    
    /**
     * Log an error message
     */
    fun e(message: String, throwable: Throwable? = null) {
        Log.e(tag, message, throwable)
    }
}
